interfaces/repo.go

```diff
@@ -283,19 +283,24 @@ func (r *Repository) Connection(connRef *ConnRef) (*Connection, error) {
 	// Ensure that such plug exists
 	plug := r.plugs[connRef.PlugRef.Snap][connRef.PlugRef.Name]
 	if plug == nil {
-		return nil, fmt.Errorf("snap %q has no plug named %q", connRef.PlugRef.Snap, connRef.PlugRef.Name)
+		return nil, &NoPlugOrSlotError{
```

Comment:
these *Error wrapping should have corresponding unit test changes, no?

Reply:
Added